src/util/primitives.js

```diff
@@ -22,6 +23,49 @@ class Ray {
         vec3.copy(out, intersection);
         return true;
     }
+
+    closestPointOnSphere(center: vec3, r: number, out: vec3): boolean {
+        assert(vec3.squaredLength(this.dir) > 0.0 && r >= 0.0);
+        if (vec3.equals(this.pos, center) || r === 0.0) {
+            out = [0.0, 0.0, 0.0];
```

Comment:
Nit: this is a noop — I suspect `out` is supposed to be modified rather than internal variable reference changed to a discarded array?

Reply:
Yes, good catch.